simple/stats/db.py

```diff
@@ -397,6 +406,34 @@ def __init__(self, db_params: dict[str, str]) -> None:
     for statement in _INIT_STATEMENTS:
       self.cursor.execute(statement)
 
+  def _maybe_create_database(connector: Connector,
+                             db_params: dict[str, str]) -> None:
+    kwargs = {
+        param: db_params[param]
+        for param in _CLOUD_MY_SQL_INSTANCE_CONNECT_PARAMS
+    }
+    db_name = db_params[CLOUD_MY_SQL_DB]
+    logging.info(
+        "Connecting to Cloud MySQL instance '%s' to check existence of DB '%s'.",
+        db_params[CLOUD_MY_SQL_INSTANCE], db_name)
+    with connector.connect(db_params[CLOUD_MY_SQL_INSTANCE], "pymysql",
+                           **kwargs) as conn:
+      with conn.cursor() as cursor:
+        # Check if the database already exists.
+        cursor.execute(
+            f"SELECT COUNT(*) FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = '{db_name}'"
+        )
+        exists = cursor.fetchone()[0] > 0
+
+        if not exists:
```

Comment:
nit: for readability, maybe a helper `_db_exists(cursor)`?

Reply:
Done.